southbound/impl/src/main/java/org/opendaylight/ansible/southbound/StatusL3vpnListener.java

```diff
@@ -0,0 +1,242 @@
+import org.opendaylight.yang.gen.v1.urn.ietf.params.xml.ns.yang.ietf.inet.types.rev130715.Ipv4Address;
+import org.opendaylight.yang.gen.v1.urn.ietf.params.xml.ns.yang.ietf.l3vpn.svc.rev170502.L3vpnSvc;
+import org.opendaylight.yang.gen.v1.urn.ietf.params.xml.ns.yang.ietf.l3vpn.svc.rev170502.Status;
+import org.opendaylight.yang.gen.v1.urn.ietf.params.xml.ns.yang.ietf.l3vpn.svc.rev170502.StatusL3vpnProvider;
+import org.opendaylight.yang.gen.v1.urn.ietf.params.xml.ns.yang.ietf.l3vpn.svc.rev170502.l3vpn.svc.fields.Sites;
+import org.opendaylight.yang.gen.v1.urn.ietf.params.xml.ns.yang.ietf.l3vpn.svc.rev170502.l3vpn.svc.fields.sites.Site;
+import org.opendaylight.yang.gen.v1.urn.ietf.params.xml.ns.yang.ietf.l3vpn.svc.rev170502.l3vpn.svc.fields.sites.site.site.network.accesses.SiteNetworkAccess;
+import org.opendaylight.yang.gen.v1.urn.ietf.params.xml.ns.yang.ietf.yang.types.rev130715.Uuid;
+import org.opendaylight.yang.gen.v1.urn.ietf.params.xml.ns.yang.l3vpn.svc.aug.rev170502.PeAugmentation;
+import org.opendaylight.yangtools.yang.binding.InstanceIdentifier;
+import org.ops4j.pax.cdi.api.OsgiService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+public class StatusL3vpnListener extends AbstractSyncDataTreeChangeListener<StatusL3vpnProvider> {
+    private static final Logger LOG = LoggerFactory.getLogger(StatusL3vpnListener.class);
+    private final RetryingManagedNewTransactionRunner txRunner;
+    private final DataBroker dataBroker;
+    private static final String COMMIT_VERSION = "123";
+    private final AnsibleCommandServiceImpl ansibleCommandService;
+    private org.opendaylight.yang.gen.v1.urn.opendaylight.ansible.command.rev180821.Status ansibleStatus;
+    private ListeningExecutorService executor;
+    private static ListenableFuture<List<Uuid>> l3vpnCommands;
+
+    @Inject
+    public StatusL3vpnListener(@OsgiService final DataBroker dataBroker,
+                               AnsibleCommandServiceImpl ansibleCommandService) {
+        super(dataBroker, LogicalDatastoreType.OPERATIONAL, STATUS_L3VPN_PATH);
+        this.txRunner = new RetryingManagedNewTransactionRunner(dataBroker, 3);
+        this.dataBroker = dataBroker;
+        this.ansibleCommandService = ansibleCommandService;
+        executor = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(5));
+    }
+
+    @Override
+    public void add(@Nonnull InstanceIdentifier<StatusL3vpnProvider> instanceIdentifier,
+                    @Nonnull StatusL3vpnProvider statusL3vpn) {
+        LOG.info("add: id: {}\nstatusL3vpn: {}", instanceIdentifier, statusL3vpn);
+        txRunner.callWithNewReadWriteTransactionAndSubmit(OPERATIONAL, tx -> {
+            try {
+                if (IetfL3vpnSvcUtils.getStatus(tx) == InProgress) {
+                    LOG.info("Creating VPN");
+                    IetfL3vpnSvcUtils.putStatusL3VPN(tx, COMMIT_VERSION, createVpn());
+                    LOG.info("VPN is being created");
```

Comment:
a lot changed to repeat ourselves

Reply:
I added this because the thread was dying in createVpn silently. I had to use log statements to figure out where the thread was dying. Talked with Stephen about this and it seems like a pretty bad flaw in ODL.